src/sequence/cache/strategies/default.js

```diff
@@ -0,0 +1,46 @@
+const { assign, omit } = require("lodash");
+
+const { sequence: makeNewSequence } = require("../../sequence");
+const { SequenceCache } = require("../sequence-cache");
+
+
+const cache = Object.create(null);
```

Comment:
Both of these are really good points...

To be honest, I'm sure there _are_ cases where a single process could potentially utilize differing cache strategies.  I attempted that at first, but it made everything much more complex, and required intermediate object instantiation that slowed down traversal.

So although its a bit icky, this seemed a reasonable first implementation.  It might be that somebody can come up with a clever way of improving the flexibility of the implementation here, or I might revisit at some point.  One thing I've considered is instantiating several values on the `Renderer` itself, and then passing that through the traversal.  That might be cleaner, and would be a good place to start if it is determined that the implementation in this PR is inadequate.

However, this does seem "good enough" in a lot of ways, since _most_ people will probably utilize a single cache backend for a given process.  And, if someone really needs to fork their behavior, they could do so in the context of the cache strategy that they register.  They wouldn't have access to every piece of information, but they would have access to the `cacheKey` and the sequence buffer.